File: scripts/Sequence/Game/Ready.lua
--- 对应 Ready.cpp
local Ready = {}

---@class Ready
local mt = {}

---@return Ready
function Ready:new()
    ---@class Ready
    local obj = {}
    setmetatable(obj, { __index = mt })
    obj.tImage = Image:new(CMAKE_SOURCE_DIR .. "res/dummy.png")
    obj.iCount = 0
    obj.bStarted = false
    return obj
end

---@param parent GameParent
---@return Ready
function mt:update(parent)
    local next = self
    if not self.bStarted then
        parent:startLoading();
        self.bStarted = true;
    end
    if self.iCount >= 120 then
    elseif self.iCount >= 60 then
        Framework:drawDebugString(1, 2, "GO!")
    else
        Framework:drawDebugString(1, 2, "Ready...")
    end
    parent:drawState();
    self.tImage:draw();
    Framework:drawDebugString(1, 1, "[READY] : ")
    self.iCount = self.iCount + 1
    return next
end

function mt:destroy()
end

return Ready

File: example/Sequence/Game/Ready.cpp
#include "GameLib/GameLib.h"
#include "Games/Framework.h"
using namespace GameLib;

#include "Sequence/Game/Ready.h"
#include "Sequence/Game/Parent.h"
#include "Image.h"

namespace Sequence{
namespace Game{

Ready::Ready() : 
mImage( 0 ), mCount( 0 ), mStarted( false ){
	mImage = new Image( CMAKE_CURRENT_SOURCE_DIR"data/image/dummy.dds" );
}

Ready::~Ready(){
	SAFE_DELETE( mImage );
}

void Ready::update( Parent* parent ){
	if ( !mStarted ){
		parent->startLoading();
		mStarted = true;
	}
	if ( mCount >= 120 ){ //2过了一秒就到Play
		parent->moveTo( Parent::NEXT_PLAY );
	}else if ( mCount >= 60 ){ //显示Go
		Framework::instance().drawDebugString( 0, 1, "GO!" );
	}else{ //显示Ready...
		Framework::instance().drawDebugString( 0, 1, "Ready..." );
	}
	//绘制
	parent->drawState();
	mImage->draw();
	//输出文字
	Framework::instance().drawDebugString( 0, 0, "[READY] : " );

	++mCount;
}

} //namespace Game
} //namespace Sequence